### PR TITLE
[Bugfix] fix retrieving wrong semester when scan

### PR DESCRIPTION
**Pull request by calvinqc**

### CODE REVIEW

File: server/.eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-    "extends": ["airbnb", "plugin:prettier/recommended"]
-  }
\ No newline at end of file
+  "extends": ["airbnb", "plugin:prettier/recommended"],
+  "rule": {
+    "no-console": "off"
```
Comment (calvinqc) [inline server/.eslintrc.json:4]: Adding rule "no-console" so it doesn't display warning for `console.log`

File: server/store/Scanning/cloudOCR.js [hunk 2]
```diff
@@ -22,12 +22,13 @@ let startingTermFound = false;
...
 let count = 0;
 let startingSem = {};
+let TranscriptMap;
 
 const resetMap = () => {
-  this.startingTermFound = false;
-  this.currentSemIndex = -1;
-  this.count = 0;
-  this.startingSem = {};
+  startingTermFound = false;
```
Comment (calvinqc) [inline server/store/Scanning/cloudOCR.js:28]: remove `this` because it's different from global variable on line ~21-25

Comment (calvinqc) [reply server/store/Scanning/cloudOCR.js:28]: This cause the problem to get wrong semesters. Sometimes it prints Fall 2021

File: server/store/Scanning/cloudOCR.js [hunk 3]
```diff
@@ -49,17 +48,23 @@ const transcriptParser = async paragraph => {
     words.push(word.symbols.map(symbol => symbol.text).join(''))
   );
 
-  const sentence = words.join(' ');
+  const removeCase = ['-', 'TOTAL', 'CUM', 'http', 'https'];
```
Comment (calvinqc) [inline server/store/Scanning/cloudOCR.js:51]: Adding this case to reduce scanning time. 

We don't use the sentence, which contains these variables

File: server/store/Scanning/cloudOCR.js [hunk 4]
```diff
@@ -49,17 +48,23 @@ const transcriptParser = async paragraph => {
...
 
+  const sentence = words.join(' ');
   switch (true) {
     // Check for Enrolling Semester
-    case semesterStatus.some(status => sentence.includes(status)):
-      TranscriptMap.semesterList[currentSemIndex].status = 1;
+    case semesterStatus.some(status => sentence.includes(status)): {
+      const currentSem = TranscriptMap.semesterList[currentSemIndex];
+      currentSem.status = 1;
+      console.log(`[In Progress Sem]`, currentSem.term, currentSem.year);
```
Comment (calvinqc) [inline server/store/Scanning/cloudOCR.js:61]: Now, I'll leave `console.log` for each case to make sure that it scans correctly.

File: server/store/utils.js [hunk 5]
```diff
@@ -316,10 +316,12 @@ export const getRemainingRequirement = courses => {
           })
       ),
     ];
-    for (const type of types) {
-      // const type = type.slice;
-      // const area = ''
-      await Requirement.find({ type })
+    for (const el of types) {
+      const splitString = el.split('-');
```
Comment (calvinqc) [inline server/store/utils.js:320]: This is work in progress for hotfix/plan #48

**Comment by ellamaolson** [comment]
How to Test: 
npm run dev
Open postman, login and do:
1. delete reset your plan
2. post Requirements 
3. post  courses 
4. post text scan of transcript

**Review by ellamaolson — APPROVED** [review]

**Review by DaleCS — APPROVED** [review]
LGTM!

**Pull request merged**